### DISCUSSION TITLE
Where Is The Comfui UI Interface code?

### DISCUSSION THREAD

**Discussion by BrechtCorbeel** [comfyanonymous/ComfyUI#3797] [General]
I get a bunch of Javascript, Python, C and Typescript when going through the base, but I can't really pin point what is enabling the overall interface, I assume it's Javascript/html and css, which I found, but I also found some compiled and not compiled C code that was named in a way that I almost suspected it to be interface related too. Is there a non compiled version of this C code?

I'm mainly looking for the code of the queue GUI and it's functionality.

**Comment by ltdrdata** [comment]
What C code did you find? ComfyUI doesn't contains C code.
The foundation of ComfyUI's front end is the Litegraph framework, and the core of the front end can be found at https://github.com/comfyanonymous/ComfyUI/tree/master/web/scripts.
https://github.com/comfyanonymous/ComfyUI/tree/master/web/extensions/core

**Reply by BrechtCorbeel** [reply]
I think it might have something to do with Cpython, but there is a bunch of .h files and .pyd that is compiled code in there.

I'll let you know in a second where I found it.

**Reply by BrechtCorbeel** [reply]
![image](https://github.com/comfyanonymous/ComfyUI/assets/68238477/243b8fa1-8c9d-43fc-9407-dab3e0faf612)
On a second look this is mainly builds from python itself
